services/genesis/templates.service.ts

```diff
@@ -0,0 +1,83 @@
+import { GlobalCore } from "@/core/module/module.types";
+import { createClient, SupabaseClient } from "@supabase/supabase-js";
+import config from "@/resources/utils/config";
+import MessageHandler from "@/core/message-handler";
+
+const messageHandler = MessageHandler.get();
+
+export interface Template {
+  id: number;
+  ownerId: string;
+  name: string;
+  createdAt: string;
+  preview: string;
+  fields: object; // Might want to define a more specific type for fields
+}
+
+export interface TemplateField {
+  type: string;
+  description: string;
+}
+
+class TemplateService {
+  private supabase: SupabaseClient;
+
+  constructor() {
+    const supabaseUrl = config.SUPABASE_URL as string;
+    const supabaseAnonKey = config.SUPABASE_API_KEY as string;
+    this.supabase = createClient(supabaseUrl, supabaseAnonKey);
+  }
+
+  async getTemplates(): Promise<Template[] | null> {
+    let { data: templates, error } = await this.supabase
+      .from("templates")
+      .select("*");
+    if (error) return messageHandler.handleError(error.message);
+    return templates as Template[];
+  }
+
+  async getTemplate(id: number): Promise<Template | null> {
+    const { data, error } = await this.supabase
+      .from("templates")
+      .select("*")
+      .eq("id", id)
```

Comment:
Here I am adding some notes: 
1. As I can see in the previous code here we are using a numeric id as the id field which is not the convention, if you see the other tables we have table id filed type set to uuid, I think it will be better to follow that convention.
2. This table should have RLS(Row Level Security) enabled otherwise any user can view or edit other users templates, but when you enable RLS, users may not be able to access the table, so we need to add a policy to that table as you can see in the [Readme](https://github.com/vocali-genesis/supabase-back-office/blob/develop/README.md#supabase-table-policies) of the develop branch on the staging. 
3. Let's make a PR with the templates table migration for the backend and delete the table from the staging once we deploy the PR to develop branch which will ad the table via the local migration that we just pushed to develop branch.